### ISSUE TITLE
[Feature Request]: Add Function `NaiveSyncBatchNorm` `1d, 2d, 3d`

### ISSUE THREAD

**Issue by rajveer43** [ml-explore/mlx#27]
### 🚀 The feature, motivation and pitch

This feature request proposes the extension of `BatchNorm1d` to the library. `NaiveSyncBatchNorm1d` is an extension of the existing `nn.BatchNorm1d` module, designed to support synchronization across multiple devices, either locally or globally. 

The motivation behind this feature request is to enhance capabilities in distributed deep learning scenarios. the proposed `NaiveSyncBatchNorm1d` module would fill this gap.

I propose the addition of the `NaiveSyncBatchNorm1d` module. The module provides the following features:
- Synchronization of batch normalization statistics (mean and variance) across multiple devices.
- Support for both global synchronization and local synchronization based on user requirements.

### Alternatives

While there are alternative ways to implement synchronized batch normalization, `NaiveSyncBatchNorm1d` offers a simple and efficient solution. 

### Additional context

Here's an example of how `NaiveSyncBatchNorm1d` can be used in PyTorch:

```python
sync_bn = NaiveSyncBatchNorm1d(num_sync_devices=4, global_sync=False, num_features=64)
output = sync_bn(input_tensor)
```

Source: https://github.com/facebookresearch/pytorchvideo/blob/64e5a17ccefcd6b93ad331d1a9c2a130f179ff44/pytorchvideo/layers/batch_norm.py#L10C44-L10C44

I would like to create a PR for the same.

**Comment by awni** [comment]
We are adding BatchNorm in #217 (closing this as a dup of #204)
